src/ocean/io/serialize/StructSerializer.d

```diff
@@ -555,9 +558,20 @@ struct StructSerializer ( bool AllowUnions = false )
                 // slice array (passing a static array as ref is not allowed)
                 U[] array = field;
 
-                static if ( is(BaseTypeOfArrays!(U) == struct) )
+                alias BaseTypeOfArrays!(U) BT;
+
+                static if ( is(BT == struct) )
                 {
-                    serializeStructArray(array, field_name, serializer, data);
+                    static if ( is(BT.IsTypedef) )
+                    {
+                        alias TypedefBaseType!(BT)[] DT;
+                        serializer.serializeArray(data, field_name, cast(DT) array,
```

Comment:
I would just instantiate it with the right template parameters and rely on implicit conversion instead of casting

Reply:
I don't think D allows implicit casting of `struct { T field; } []` to `T[]` , does it?

Reply:
I don't quite get what you mean ;)